### ISSUE TITLE
keyword_quote_linter() could allow quoted names in "balanced expressions"

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2055]
Follow-up to #2030.

One thing we see with some regularity is usage like

```r
switch(x,
  "a" = 1,
  "b c" = 2,
  "d" = 3,
  "e f" = 4
)
```

Here, the quotes aren't strictly necessary, but can help with readability/consistency.

I would allow such usage by default, and not introduce an option allowing the more strict behavior (hence labeling this `false-positive`), but we could consider adding such an option as well.
